zwad/ad/__init__.py

```diff
@@ -88,6 +76,10 @@ def run(self):
         if self.args.output:
             scores.tofile(self.args.output)
 
+    @abstractmethod
+    def _load_dataset(self):
+        raise NotImplementedError
```

Comment:
There is no need for using both `@abstractmethod` and `NotImplementedError`. IIRC, `NotImplementedError` was introduced as a plug for abstract methods when there were no abc-classes in Python.